pinot-core/src/main/java/org/apache/pinot/core/io/writer/ForwardIndexWriter.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.io.writer;
+
+import java.io.Closeable;
+
+
+/**
+ * Interface for forward index writer.
+ */
+public interface ForwardIndexWriter extends Closeable {
+
+  /**
+   * SINGLE-VALUE COLUMN APIs
+   */
+
+  /**
+   * Writes the INT type single-value into the given document id.
+   * <p>NOTE: Dictionary id is handled as INT type.
+   *
+   * @param docId Document id
+   * @param value Value to write
+   */
+  default void setInt(int docId, int value) {
```

Comment:
If this is a new interface, why provide `default` here (as opposed to an abstract base class)?

Reply:
The difference between default methods in interface vs an common abstract base class:
- A class cannot extend multiple base classes but can implement multiple interfaces
- The interface extending this (e.g. ForwardIndexReaderWriter) can automatically get the default implementation
- IMO this is more readable than a common base class that all classes extend